src/main/thrift/parquet.thrift

```diff
@@ -578,7 +578,13 @@ enum BoundaryOrder {
 
 /** Data page header */
 struct DataPageHeader {
-  /** Number of values, including NULLs, in this data page. **/
+  /**
+   * Number of values, including NULLs, in this data page.
+   *
+   * If a ColumnIndex is present, a page must begin at a record
```

Comment:
Should we change to "PageIndex"

Reply:
That is an excellent question @mapleFU  -- given the discussion on https://github.com/apache/parquet-format/pull/245 I would prefer to refer to the definitions of ColumnIndex and OffsetIndex instead.

So perhaps something like this 🤔 

```suggestion
   * If a ColumnIndex or OffsetIndex is present, a page must begin at a record
```

Reply:
Yeah, they all lgtm 🤔

Reply:
I also asked for clarification on the mailing list: https://lists.apache.org/thread/rd8twnvg4bg3558r507rzpxckcxt5wdn

Upon further thought it seems like since `OffsetIndex` is a pointer from records to pages, then maybe this is more accurate:

```suggestion
   * If a OffsetIndex is present, a page must begin at a record
```